tests/main.test.ts

```diff
@@ -47,22 +47,66 @@ describe("src/main.ts", function() {
 
     const validAny = validModel(anyType);
 
-    it('should allow all types when no restriction', function(done) {
+    it('should allow all types for -any- schema when there are no restriction',
+    function(done) {
       const validTypes = [
         undefined,
         null,
         5,
         "string",
         true,
-        {first: 1, second: "two"},
-        // null,
-        // undefined
+        {first: 1, second: "two"}
       ];
 
-      mochaAssertPromiseResovles(
-        Promise.all(validTypes.map(validAny)),
+      mochaAssertPromiseResovles(Promise.all(validTypes.map(validAny)), done);
+    });
+
+    const anyWithBoolRestriction: anySchema = {
+      isAny: true,
+      nullAllowed: true,
+      undefinedAllowed: true,
+      restriction: (someBool: boolean) => {
+        if(someBool) {
+          return Promise.reject("error");
+        }
+      }
+    };
+
+    const validAnyWithBoolRestriction = validModel(anyWithBoolRestriction);
+
+    it('should fail against the restriction', function(done) {
+      mochaAssertPromiseErrorsWith(
+        validAnyWithBoolRestriction(true),
+        (error) => {
+          return error === "error"
+        },
         done
-      );
+      )
+    });
+
+    const anyWithLengthRestriction: anySchema = {
+      isAny: true,
+      nullAllowed: true,
+      undefinedAllowed: true,
+      restriction: (arrayOfNumber: number[]) => {
```

Comment:
The problem with this and your `bool` one above is that you're naming it `arrayOfNumber` or `someBool` when it's not just arrays/bools that would work for it. It would be confusing that  `{ length: 5 }` passed a restriction designed for arrays.

The restriction isn't actually designed for arrays though, so let's have the naming reflect that. So right now where you do `arrayOfNumber: number[]` it would make more sense to do `anyWithLength: { length: number }` because you're restriction is validating anything with a length.

Also keep in mind that the *point* of the `anySchema` was to be able to validate different types, so it'd be nice to have a validator that validates `[1,2,3,4]` and `{ length: 4 }` because we made the `anySchema` to be able to handle these situations.